### PR TITLE
add a job to PR workflow

### PR DESCRIPTION
**Pull request by bernt-matthias**
that indicates success if no tools were changed

@mvdbeek maybe something like this helps?

FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Review by mvdbeek — APPROVED** [review]
Yeah, I think that would help, although I also like being able to just override the checks.
We probably need to merge this to see it in effect.

**Comment by mvdbeek** [comment]
Ah, I think this needs to be a single status check that reflects no tools and successful checks:
Screenshot 2021-05-13 at 16 17 31
I don't think we can specify an `or` here

**Comment by bernt-matthias** [comment]
Maybe something like:

```
success-if-notools:
    if: always()
    needs: [setup, combine]
```

and then implement the logic in a step.. not sure how.

**Comment by bernt-matthias** [comment]
Checked here https://github.com/bernt-matthias/tools-iuc/pull/37 (https://github.com/bernt-matthias/tools-iuc/runs/2577704110) .. with the badges PR

**Comment by mvdbeek** [comment]
Nice!

**Review by mvdbeek — APPROVED** [review]

**Comment by nsoranzo** [comment]
Wouldn't adding:
```
    if: needs.setup.outputs.repository-list != ''
```
to the `combine` step be enough?

BTW, I believe `if: needs.setup.outputs.repository-list != ''` should instead be removed from the `lint` step, otherwise it won't check for missing `.shed.yml` file for a new tool.

**Comment by bernt-matthias** [comment]
> Wouldn't adding:
> 
> ```
>     if: needs.setup.outputs.repository-list != ''
> ```
> 
> to the `combine` step be enough?

What if the combine step fails?

> BTW, I believe if: needs.setup.outputs.repository-list != '' should instead be removed from the lint step, otherwise it won't check for missing .shed.yml file for a new tool.

Either this or `needs.setup.outputs.repository-list != '' and needs.setup.outputs.tool-list != ''`. Not sure. We can try to remove it.

**Comment by nsoranzo** [comment]
> > Wouldn't adding:
> > ```
> >     if: needs.setup.outputs.repository-list != ''
> > ```
> >   
> > to the `combine` step be enough?
> 
> What if the combine step fails?

Aren't you trying to workaround a failure when no tools were modified? In which case the combine step should be skipped with the above condition.

> > BTW, I believe if: needs.setup.outputs.repository-list != '' should instead be removed from the lint step, otherwise it won't check for missing .shed.yml file for a new tool.
> 
> Either this or `needs.setup.outputs.repository-list != '' and needs.setup.outputs.tool-list != ''`. Not sure. We can try to remove it.

We weren't skipping the `lint` step before the move to `planemo-ci-action` , so I suppose it should still work.

**Comment by bernt-matthias** [comment]
> Aren't you trying to workaround a failure when no tools were modified? In which case the combine step should be skipped with the above condition.

Not sure if a skipped combine step would count as successful with the settings outlined by @mvdbeek https://github.com/galaxyproject/tools-iuc/pull/3669#issuecomment-840593344

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -335,3 +334,25 @@ jobs:
...
         shed-key: ${{ secrets.TS_API_KEY }}
+
+  determine-success:
+    name: Check workflow success
+    runs-on: ubuntu-latest
+    if: always()
+    needs: [setup, lint, flake8, lintr, combine_outputs]
+    strategy:
+      matrix:
+        python-version: ['3.7']
```
Comment (nsoranzo) [inline .github/workflows/pr.yaml:345]:
```suggestion
```

**Comment by nsoranzo** [comment]
> > Aren't you trying to workaround a failure when no tools were modified? In which case the combine step should be skipped with the above condition.
> 
> Not sure if a skipped combine step would count as successful with the settings outlined by @mvdbeek [#3669 (comment)](https://github.com/galaxyproject/tools-iuc/pull/3669#issuecomment-840593344)

It does, I've just played around with those settings and requiring the "Deploy" step (which is always skipped on PRs) still allows merging.

I've now changed the settings to require "Lint tool-list (3.7)", "Lint Python scripts (3.7)", "Lint R scripts (ubuntu-20.04, 4.0.1)" and "Combine chunked test results (3.7)" (which can still force the merge as admins).

**Comment by mvdbeek** [comment]
I'm a bit confused by this.

> It does, I've just played around with those settings and requiring the "Deploy" step (which is always skipped on PRs) still allows merging.

That was the starting point, and it definitely did not allow us merging a PR that skipped a required test without changing the settings allowing admins to merge. Which we should avoid, not all committers are or should be repo admins.

So you're saying that this is now not necessary anymore ?

**Comment by nsoranzo** [comment]
> I'm a bit confused by this.
> 
> > It does, I've just played around with those settings and requiring the "Deploy" step (which is always skipped on PRs) still allows merging.
> 
> That was the starting point, and it definitely did not allow us merging a PR that skipped a required test without changing the settings allowing admins to merge. Which we should avoid, not all committers are or should be repo admins.
> 
> So you're saying that this is now not necessary any more ?

The starting point was not the "Deploy" step, it was the "Combine chunked test results (3.7)" , xref. https://github.com/galaxyproject/tools-iuc/pull/3669#issuecomment-840593344

**Comment by nsoranzo** [comment]
OK, "interestingly" the job name changes by losing the matrix part if the job is skipped, see e.g. the difference between "Lint tool-list (3.7)" (executed) and "Lint Python scripts" (skipped) in the box below.
This discrepancy leaves the status check waiting for non-existing jobs in both following situations:
- if no tools were changed (as in this PR) and the status check configuration uses job names *with* the matrix part
- if tools were changed in a PR and the status check configuration uses job names *without* the matrix part

So I had to restore the `determine-success` job, with a couple of fixes. I also found such a solution proposed in https://github.community/t/status-check-for-a-matrix-jobs/127354

**Review by nsoranzo — APPROVED** [review]

**Comment by bernt-matthias** [comment]
Looks good to me. Thanks for the feedback @nsoranzo and @mvdbeek

**Comment by bernt-matthias** [comment]
Seems to work: https://github.com/galaxyproject/tools-iuc/pull/3636

One could think about running `lint` only `if: ${{ needs.setup.outputs.repository-list != '' && needs.setup.outputs.tool-list != '' }}`

**Pull request merged**